src/app/components/shop-list/shop-list.component.ts

```diff
@@ -0,0 +1,130 @@
+import { Component } from '@angular/core';
+import { ListproductsService } from '../../services/listproducts.service';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { RouterModule, ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-shop-list',
+  standalone: true,
+  imports: [CommonModule, FormsModule, RouterModule],
+  templateUrl: './shop-list.component.html',
+  styleUrls: ['./shop-list.component.css']
+})
+export class ShopListComponent {
+  products: any[] = [];
+  filteredProducts: any[] = [];
+  categories: string[] = [];
+  searchTerm: string = ''; 
+  selectedCategory: string = ''; 
+  displayCount: number = 6;
+
+  constructor(
+    private listproductsService: ListproductsService, 
+    private router: Router, 
+    private route: ActivatedRoute 
+  ) {}
+
+  ngOnInit(): void {
+    this.listproductsService.getProducts().subscribe({
+      next: (data) => {
+        this.products = data;
+        this.filteredProducts = data; 
+        this.extractCategories(data);
+
+        this.route.queryParams.subscribe(params => {
+          const category = params['category'];
+          const search = params['search'];
+
+          if (category) {
+            this.selectedCategory = category;
+          }
+
+          if (search) {
+            this.searchTerm = search;
+          }
+
+          this.applyFilters(); 
+        });
+      },
+      error: (err) => {
+        console.error('Error al obtener los productos:', err);
+      }
+    });
+  }
+
+  extractCategories(products: any[]) {
+    const categoriesSet = new Set(products.map(product => product.category));
+    this.categories = Array.from(categoriesSet);
+  }
+
+  filterByCategory(event: Event) {
```

Comment:
There is an endpoint in the back that receives a query param to filter by category.

Reply:
I just fixed, please review again